js/client-battle.js

```diff
@@ -1002,18 +997,23 @@
 				text += '</div></div>';
 				break;
 
-			case 'pokemon':
-				var pokemon = this.battle.getPokemon(thing);
+			case 'pokemon': // Tooltip for active pokemon. Initially we select the spectator/opponent sees view.
+				var side = this.battle[thing.slice(0, -1) + "Side"];
+				var pokemon = side.active[thing.slice(-1)];
 				if (!pokemon) return;
 				/* falls through */
-			case 'sidepokemon':
+			case 'sidepokemon': // Tooltip for switching pokemon; myPokemon contains the latest server state.
 				var battlePokemon;
 				var myPokemon;
 				if (this.myPokemon) {
 					if (!pokemon) {
+						// We didn't fall through, so we're switching. Just show the server state.
 						pokemon = this.myPokemon[parseInt(thing)];
 						battlePokemon = this.battle.getPokemon('other: old: ' + pokemon.ident, pokemon.details);
 					} else if (pokemon.side === this.battle.mySide) {
```

Comment:
Sorry, I'm not sure what you mean there. There are several cases:
- Choosing one of your own Pokémon for a switch (or targeting an ally in doubles/triples). This uses the `'sidepokemon'` case, and controlsShown is always true. This case uses the noted line and none of the other cases do.
- Targeting one of your opponent's Pokémon in doubles/triples. This uses the `'pokemoin'` case, and controlsShown is always true, but it's your opponent, so it makes no difference.
- Hovering over your opponent's Pokémon in the battle. This uses the `'pokemon'` case, and controlsShown may or may not be true, but it's your opponent, so it makes no difference.
- Hovering over your own Pokémon in the battle. This uses the `'pokemon'` case, and controlsShown may or may not be true. The `else if` part of the noted condition takes effect allowing the stats to be shown (and the ability even if the opponent doesn't know it). The current code assumes that if the opponent knows your Ability (e.g. due to in-turn activation, Skill Swap, Entrainment, Worry Seed etc.) then that's always at least as up-to-date as the server, so it doesn't check `controlsShown`.